include/RAJA/policy/desul/atomic.hpp

```diff
@@ -26,153 +25,208 @@ using raja_default_desul_scope = desul::MemoryScopeDevice;
 namespace RAJA
 {
 
+namespace detail
+{
+template <typename T>
+struct DesulAtomicPolicy {
+  using memory_order = raja_default_desul_order;
+  using memory_scope = raja_default_desul_scope;
+};
+
+template <typename OrderingPolicy, typename ScopePolicy>
+struct DesulAtomicPolicy<detail_atomic_t<OrderingPolicy, ScopePolicy>> {
+  using memory_order = OrderingPolicy;
+  using memory_scope = ScopePolicy;
+};
+
+}  // namespace detail
+
 RAJA_SUPPRESS_HD_WARN
 template <typename AtomicPolicy, typename T>
-RAJA_HOST_DEVICE
-RAJA_INLINE T
-atomicAdd(AtomicPolicy, T volatile *acc, T value) {
-  return desul::atomic_fetch_add(const_cast<T*>(acc),
+RAJA_HOST_DEVICE RAJA_INLINE T atomicAdd(AtomicPolicy, T volatile *acc, T value)
+{
+  using desul_order =
+      typename detail::DesulAtomicPolicy<AtomicPolicy>::memory_order;
```

Comment:
Does this mean that AtomicPolicy has to be `detail_atomic_t<...>` instead of `DesulAtomicPolicy<...>`